Add text split node

Moult opened the pull request:
## Addressed problem description

In #3565 it was identified that text nesting was inconsistent, and that the Note node actually implicitly split text. As part of the proposal to standardise text nesting, I also suggested that for those who preferred the old behaviour of split text, a new split node would be added. As seen in #3618 - users need this split node :) so here it is!

## Solution description

In the old Note node, if you typed `'foo bar'`, it would come out as `[[['foo', 'bar']]]`. Now it comes out as `[['foo bar']]`. If you then apply this new explicit split mode, it will split (by whitespace, but maybe we can add a configurable delimiter in the future?) into the original behaviour.

## Preflight checklist

Put an x letter in each brackets when you're done this item:

- [x] Code changes complete.
- [x] Code documentation complete. (Not sure what to document, quite straightforward)
- [ ] Documentation for users complete (Will add if people are happy with behaviour).
- [x] Manual testing done. 
- [ ] Unit-tests implemented. (What's the process for this?)
- [ ] Ready for merge.

Moult commented:
![image](https://user-images.githubusercontent.com/88302/95280030-c4d44500-089f-11eb-9642-6ce1a6e33261.png)

Durman commented:
What about just adding option to the `Note` node which will switch between old and new string formants?
For me what should do `text split` node is a little bit unobvious.

Moult commented:
The advantage is an explicit split node can be used on any text input, not just note.

Split effectively does `.split()` in Python. When parsing external datasets this is quite useful.

nortikin commented:
maybe make string manipulation nodes stack? `join`/`split`/`to_string`/`evaluate_from_string` ...
+ `list pop node` needed also for lists.

vicdoval commented:
I would be nice if you could set the separation character, space, coma, user defined...

Moult commented:
@nortikin agree the full stack is needed. I will do in a separate PR, as this one has a higher priority since users are waiting for it.

@vicdoval yes, agree.

Can somebody guide me if there is anything I should be aware of on creating a unit test for this, as well as how nodes are documented? If people are happy with the implementation I'll add test / documentation (and add the delimiter feature too).

portnov commented:
We do not have a very good technology for testing *nodes*. What we do have is something like https://github.com/nortikin/sverchok/blob/master/tests/simple_generator_tests.py. It looks a bit heavy: the test will programmatically create the node tree and so on. I think, general idea should be that all "interesting" functionality should be enclosed in separate functions / modules, and tests should test those functions, not nodes. There will be some uncovered functionality that remained in the node class, but that should be more or less "trivial" (like data matching).

rendetto commented:
Umm... what is the state of this node. Will it be in master any time soon? For me it works without problems, and it is essential node in SV.

vicdoval commented:
@rendetto for the moment you can use this
![image](https://user-images.githubusercontent.com/10011941/103294407-1c430680-49f2-11eb-9137-a0277f805ce3.png)

rendetto commented:
Thank you @vicdoval !

zeffii commented:
i think this can be closed, the `String Tools` node encapsulates this behaviour.

![image](https://user-images.githubusercontent.com/619340/116004011-1f957a00-a601-11eb-812d-21c6b9830abd.png)
